lit/docs/guides/git.lit

```diff
@@ -139,3 +139,230 @@
+  }{
+    Automatically \reference{multi-branch-cleaning-up} after branches get
+    merged or deleted
+  }
+
+  \section{
+    \title{Test, Build & Deploy}{multi-branch-test-build-deploy}
+
+    We'll start out by defining the pipeline that should run for each active
+    branch. For this example, we'll be working with the following \link{sample Go
+    application}{https://github.com/concourse/examples/tree/master/apps/golang}.
+
+    Our pipeline will have three stages:
+
+    \ordered-list{
+      Run unit tests
+    }{
+      Build and upload a binary to a blobstore (in our case, we'll use
+      \link{Google Cloud Storage}{https://cloud.google.com/storage})
+    }{
+      Trigger a \code{terraform apply} to deploy our app to a staging
+      environment. The \link{Terraform module}{https://github.com/aoldershaw/examples/blob/master/terraform/staging/main.tf}
+      we'll use here doesn't actually provision any infrastructure, and is just used
+      as an example
+    }
+
+    Since the pipeline config is intended to be used as a template for multiple
+    different branches, we can use \reference{vars} to parameterize the config.
+    In particular, we'll use the vars \code{((feature))} and \code{((branch))},
+    which represent the name of the feature and the name of the branch, respectively.
+
+    Below is the full pipeline config:
+
+    \link{\code{examples/pipelines/multi-branch/template.yml}}{https://github.com/aoldershaw/examples/blob/master/pipelines/multi-branch/template.yml}
+    \remote-codeblock{yaml}{https://raw.githubusercontent.com/aoldershaw/examples/master/pipelines/multi-branch/template.yml}
+  }
+
+  \section{
+    \title{Tracking Branches}{multi-branch-tracking-branches}
+
+    In addition to the branch pipeline template, we'll also need a pipeline to
+    track the list of branches and set a pipeline for each one.
+
+    To track the list of branches in a repository, we can use
+    \link{aoldershaw/git-branches-resource}{https://github.com/aoldershaw/git-branches-resource}.
+    This \reference{resource-types}{\code{resource_type}} emits a new
+    \reference{resource-versions}{resource version} whenever a branch is created or
+    deleted. It also lets us filter the list of branches by a regular expression.
+    In this case, let's assume our feature branches match the regular expression
+    \code{feature/.*}.
+
+    Below is the full pipeline config for this tracker pipeline:
+
+    \link{\code{examples/pipelines/multi-branch/tracker.yml}}{https://github.com/aoldershaw/examples/blob/master/pipelines/multi-branch/tracker.yml}
+    \codeblock{yaml}{{{
+      resource_types:
+      - name: git-branches
+        type: registry-image
+        source:
+          repository: aoldershaw/git-branches-resource
+
+      resources:
+      - name: feature-branches
+        type: git-branches
+        source:
+          uri: https://github.com/concourse/examples
+          branch_regex: 'feature/(?P<feature>.*)'
```

Comment:
What's up with this regex?
<img width="279" alt="image" src="https://user-images.githubusercontent.com/4583837/126389855-b5bb07bd-4498-4c0d-bbc7-fe345b53bb5d.png">
```suggestion
          branch_regex: 'feature/.*'
```

Reply:
Ah, should probably explain that a bit better. It's a named capture group in a few language's regex implementations (at least Python and Go). https://github.com/aoldershaw/git-branches-resource uses named capture groups to capture a portion of the branch name for use in setting the pipeline (note the `((.:branch.groups.feature))`, which is using this captured value, omitting the leading `feature/`)